src/librustc_mir/build/mod.rs

```diff
@@ -687,7 +689,9 @@ fn construct_const<'a, 'tcx>(
     let mut block = START_BLOCK;
     let ast_expr = &tcx.hir().body(body_id).value;
     let expr = builder.hir.mirror(ast_expr);
-    unpack!(block = builder.into_expr(&Place::return_place(), block, expr));
+    // We don't provide a scope because we can't panic in constants, so won't
```

Comment:
we can't unwind, though we can in fact panic (but it's always treated as abort-on-panic or in fact error-on-panic)